frontend/app/routes/protected/person-case/primary-docs.tsx

```diff
@@ -44,45 +52,166 @@ export async function action({ context, request }: Route.ActionArgs) {
 
-  if (formData.get('action') === 'back') {
-    throw i18nRedirect('routes/protected/person-case/privacy-statement.tsx', request); //TODO: change it to redirect to file="routes/protected/person-case/request-details.tsx"
-  }
+  switch (action) {
+    case 'back': {
+      throw i18nRedirect('routes/protected/index.tsx', request);
+    }
 
-  // submit action
-  const schema = v.object({
-    currentStatusInCanada: v.pipe(
-      v.string(t('protected:primary-identity-document.current-status-in-canada.required')),
-      v.trim(),
-      v.nonEmpty(t('protected:primary-identity-document.current-status-in-canada.required')),
-      v.literal(
-        'canadian-citizen-born-outside-canada',
-        t('protected:primary-identity-document.current-status-in-canada.invalid'),
-      ),
-    ),
-  });
+    case 'next': {
+      const input: { currentStatusInCanada?: string; documentType?: string } = {};
 
-  const input = { currentStatusInCanada: formData.get('currentStatusInCanada') as string };
-  const parsedDataResult = v.safeParse(schema, input, { lang });
+      const currentStatusSchema = v.object({
+        currentStatusInCanada: v.pipe(
+          v.string(t('protected:primary-identity-document.current-status-in-canada.required')),
+          v.trim(),
+          v.nonEmpty(t('protected:primary-identity-document.current-status-in-canada.required')),
+          v.picklist(VALID_CURRENT_STATUS, t('protected:primary-identity-document.current-status-in-canada.invalid')),
+        ),
+      });
+      input.currentStatusInCanada = formData.get('currentStatusInCanada') as string;
+      const currentStatusResult = v.safeParse(currentStatusSchema, input, { lang });
 
-  if (!parsedDataResult.success) {
-    return data({ errors: v.flatten<typeof schema>(parsedDataResult.issues).nested }, { status: 400 });
-  }
+      let documentTypeResult = null;
 
-  context.session.inPersonSINCase = {
-    ...(context.session.inPersonSINCase ?? {}),
-    ...input,
-  };
+      if (formData.get('step') === '2') {
+        const documentTypeSchema = v.object({
+          documentType: v.pipe(
+            v.string(t('protected:primary-identity-document.document-type.required')),
+            v.trim(),
+            v.nonEmpty(t('protected:primary-identity-document.document-type.required')),
+          ),
+        });
+        input.documentType = formData.get('documentType') as string;
+        documentTypeResult = v.safeParse(documentTypeSchema, input, { lang });
+
+        if (!documentTypeResult.success) {
+          return data(
+            {
+              errors: {
+                ...(currentStatusResult.issues ? v.flatten<typeof currentStatusSchema>(currentStatusResult.issues).nested : {}),
+                ...v.flatten<typeof documentTypeSchema>(documentTypeResult.issues).nested,
+              },
+            },
+            { status: 400 },
+          );
+        }
+      } else if (!currentStatusResult.success) {
```

Comment:
Bit of a nitpick, but this if/else could be simplified a bit by moving the `else if` up and returning early.. I think this would improve readability.

ex:

``` typescript
if (!currentStatusResult.success) {
  return data({ errors: v.flatten<typeof currentStatusSchema>(currentStatusResult.issues).nested }, { status: 400 });
}

// rest of function without the need for "else"
```